nexus/db-model/src/virtual_provisioning_collection.rs

```diff
@@ -0,0 +1,57 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::schema::virtual_provisioning_collection;
+use uuid::Uuid;
+
+#[derive(Debug)]
+pub enum CollectionTypeProvisioned {
+    Project,
+    Organization,
+    Silo,
+    Fleet,
+}
+
+impl std::fmt::Display for CollectionTypeProvisioned {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            CollectionTypeProvisioned::Project => write!(f, "project"),
+            CollectionTypeProvisioned::Organization => {
+                write!(f, "organization")
+            }
+            CollectionTypeProvisioned::Silo => write!(f, "silo"),
+            CollectionTypeProvisioned::Fleet => write!(f, "fleet"),
+        }
+    }
+}
+
+/// Describes virtual_provisioning_collection for a collection
+#[derive(Clone, Selectable, Queryable, Insertable, Debug)]
+#[diesel(table_name = virtual_provisioning_collection)]
+pub struct VirtualProvisioningCollection {
+    pub id: Uuid,
+    pub collection_type: String,
+
+    pub virtual_disk_bytes_provisioned: i64,
```

Comment:
Done in 76704e7fd7f9bcfefafd8de3d057cb09562d2770 for the things that can be `ByteCount`. `InstanceCpuCount`, unfortunately, can only be up to a u16, so that does not really work too well for these larger aggregate numbers - I'm planning on leaving that one as an i64 for now.

If we really want, I could make an `AggregateInstanceCpuCount`, but that might only be used in this spot?

Reply:
Up to you.  "Bytes" is easier to misuse so I think it benefits more from the newtype than the cpu count would.

Reply:
Yeah, I'm fine keeping `CpuCount` as an i64 then